Rocket.Chat/Controllers/Subscriptions/SubscriptionsViewController.swift

```diff
@@ -125,15 +121,42 @@ final class SubscriptionsViewController: BaseViewController {
     }
 }
 
+// MARK: Side Menu callbacks
+extension SubscriptionsViewController {
+    func willHide() {
+        self.textFieldSearch.resignFirstResponder()
+    }
+
+    func didHide() {
+        self.textFieldSearch.resignFirstResponder()
+    }
+
+    func willReveal() {
+        if searchText.isEmpty {
+            hideCancelSearchButton()
+        } else {
+            showCancelSearchButton()
+        }
+
+        self.textFieldSearch.resignFirstResponder()
+        self.updateData()
+    }
+
+    func didReveal() {
+        self.textFieldSearch.resignFirstResponder()
+    }
+}
+
 extension SubscriptionsViewController {
 
     @IBAction func buttonCancelSearchDidPressed(_ sender: Any) {
         textFieldSearch.resignFirstResponder()
+        textFieldSearch.text = ""
     }
 
     func searchBy(_ text: String = "") {
         guard let auth = AuthManager.isAuthenticated() else { return }
-        subscriptions = auth.subscriptions.filter("name CONTAINS %@", text)
+        subscriptions = auth.subscriptions.filterBy(name: searchText)
```

Comment:
`searchText` should be `text` here, right?

Reply:
Copy and paste is my nemesis!!